src/Microsoft.AspNet.Mvc.Core/Internal/Routing/LinkGenerationDecisionTree.cs

```diff
@@ -123,24 +125,29 @@ public AttributeRouteLinkGenerationEntryClassifier()
             }
         }
 
-        private class AttributeRouteLinkGenerationEntryComparer : IComparer<AttributeRouteLinkGenerationEntry>
+        private class LinkGenerationMatchComparer : IComparer<LinkGenerationMatch>
```

Comment:
nit: might be worth a reminder that lower is "better" (and therefore sorted earlier) for all properties checked in this comparer